tools/asset-inventory/asset_inventory/api_schema.py

```diff
@@ -145,37 +213,49 @@ def _properties_map_to_field_list(cls, properties_map, resources,
             else:
                 field['mode'] = 'NULLABLE'
             if bigquery_type == 'RECORD':
-                property_properties_map = cls._get_properties_map_from_value(
+                fields_list = cls._get_properties_map_field_list(
                     property_name, property_value, resources, seen_resources)
-                if not property_properties_map:
-                    continue
-                fields_list = cls._properties_map_to_field_list(
-                    property_properties_map, resources, seen_resources)
                 if not fields_list:
                     continue
                 field['fields'] = fields_list
             fields.append(field)
         return fields
 
+    @classmethod
+    def _get_cache_key(cls, resource_name, document):
+        if 'id' in document:
+            return '{}.{}'.format(document['id'], resource_name)
+        if 'info' in document:
+            info = document['info']
+            return '{}.{}.{}'.format(info['title'],
+                                     info['version'],
+                                     resource_name)
+        return resource_name
+
+    @classmethod
+    def _get_document_resources(cls, document):
+        if 'schemas' in document:
```

Comment:
nit: use type specific methods for better error handling in case someone tries to modify this.
We can do the same using the `dict.get(key)` which returns the value or `None` which makes it clear that document should be dict. 

Currently this will throw an error on the next line saying str or list indices must be int not str.
With the proposed change it will throw an error that says str objects has no method get. 
IMHO this is a clearer error.
```suggestion
        if document.get('schemas'):
```

Reply:
done